bitriseio/bitriseio.go

```diff
@@ -1,49 +1,59 @@
 package bitriseio
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
 	"os"
-	"regexp"
-	"strings"
 
+	"github.com/bitrise-io/codesigndoc/bitriseio/bitrise"
+	"github.com/bitrise-io/codesigndoc/models"
+	"github.com/bitrise-io/codesigndoc/utility"
 	"github.com/bitrise-io/go-utils/colorstring"
 	"github.com/bitrise-io/go-utils/log"
 	"github.com/bitrise-io/go-utils/sliceutil"
+	"github.com/bitrise-io/go-xcode/certificateutil"
 	"github.com/bitrise-io/goinp/goinp"
-	"github.com/bitrise-tools/codesigndoc/bitriseio/bitrise"
-	"github.com/bitrise-tools/go-xcode/certificateutil"
-	"github.com/bitrise-tools/go-xcode/profileutil"
 )
 
-// UploadCodesigningFiles ...
-func UploadCodesigningFiles(certificates []certificateutil.CertificateInfoModel, profiles []profileutil.ProvisioningProfileInfoModel, certsOnly bool, outputDirPath string) (bool, bool, error) {
+// GetInteractiveConfigClient asks for access token and app, returns a bitrise client
+func GetInteractiveConfigClient() (*bitrise.Client, error) {
 	accessToken, err := askAccessToken()
 	if err != nil {
-		return false, false, err
+		return nil, err
 	}
 
-	bitriseClient, appList, err := bitrise.NewClient(accessToken)
+	client, err := bitrise.NewClient(accessToken)
 	if err != nil {
-		return false, false, err
+		return nil, err
+	}
+
+	appList, err := client.GetAppList()
+	if err != nil {
+		return nil, err
 	}
 
 	selectedAppSlug, err := selectApp(appList)
 	if err != nil {
-		return false, false, err
+		return nil, err
 	}
+	client.SetSelectedAppSlug(selectedAppSlug)
 
-	bitriseClient.SetSelectedAppSlug(selectedAppSlug)
+	return client, nil
+}
 
+// UploadCodesigningFiles ...
+func UploadCodesigningFiles(client *bitrise.Client, certificates models.Certificates, profiles []models.ProvisioningProfile) (bool, bool, error) {
 	var provProfilesUploaded bool
-	if !certsOnly {
-		provProfilesUploaded, err = uploadExportedProvProfiles(bitriseClient, profiles, outputDirPath)
+	if len(profiles) != 0 {
```

Comment:
what about the `certsOnly` mode?

Reply:
In that case len(profiles) should equal 0.